### PR TITLE
pine64-pinephonepro audio support

### PR DESCRIPTION
**Pull request by wentam**
Audio tested working with this addition

### CODE REVIEW

**Comment by samueldr** [comment]
Thanks, I'm looking into refreshing everything Mobile NixOS today, so I'll be testing this in addition to updating everything.

File: devices/pine64-pinephonepro/sound.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#
+# A hack to avoid mass rebuilds of packages that depend on
+# alsaLib.
+
+{ config, lib, ... }:
+
+let
+  # From https://gitlab.com/pine64-org/pine64-alsa-ucm
+  ucm2 = "${./ucm2}";
+in {
```
Comment (samueldr) [inline devices/pine64-pinephonepro/sound.nix]: We should download the files, e.g. `fetchFromGitLab` rather than copying them locally.

Comment (wentam) [reply devices/pine64-pinephonepro/sound.nix]: Good point. Was just modeling this off of pine64-pinephone. Also don't need to worry about license that way.

**Comment by wentam** [comment]
--amend'ed with a better approach: This gitlab repo can be reused for the pinephone if desired as the repo contains the configuration for both.

As such, this uses an overlay package that contains our extra needed config/patch, and could be re-used for the pinephone in it's own device config. We need a derivation anyway for our patch/config.

I consider this hack still, as long-term I'd like to see everything upstream in https://github.com/alsa-project/alsa-ucm-conf

Edit: I also don't this patch is the right approach, as the files that are referenced do exist in alsa-ucm-conf repo - but it is working as-is.

**Comment by wentam** [comment]
Now the pine64-alsa-ucm derivation symlinks the relevant components from pkgs.alsa-ucm-conf, and follows the same directory structure as alsa-ucm-conf.

To do this right, that reference is unavoidable - because the PinePhonePro.conf file references components of alsa-ucm-conf.

Should theoretically work with the non-pro pinephone too, but I don't have one to test.

PinePhonePro is retested working with the new setup.

**Pull request closed without merging**